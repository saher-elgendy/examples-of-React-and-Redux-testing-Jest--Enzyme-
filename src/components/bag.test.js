import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Bag from './bag';
import './../setupTests';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);

describe('BookStore component', () => {
    let wrapper;
    let store;
    let mockProps = {
        bag: []
    }
    beforeEach(() => {
        mockProps = {
            bag: [],
        }

        store = mockStore({
            bag: []
        })
        wrapper = mount(
            <Provider store={store}>
                <Bag {...mockProps} />
            </Provider>
        )
    })

    it('render BookStore component correctly', () => {
        expect(mountToJson(wrapper)).toMatchSnapshot();
    });

    it('toggle bag show when click the bag image', () => {
        const image = wrapper.find('#bag-image')
        image.simulate('click');
        let bagContent = wrapper.find('#bag-content');
        expect(bagContent.exists()).toBeTruthy();
        image.simulate('click');
        bagContent = wrapper.find('#bag-content');
        expect(bagContent.exists()).toBeFalsy();
    })
})