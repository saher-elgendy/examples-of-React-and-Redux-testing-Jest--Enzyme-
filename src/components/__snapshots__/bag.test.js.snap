// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookStore component render BookStore component correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Bag)
    bag={Array []}
  >
    <Bag
      bag={Array []}
      dispatch={[Function]}
    >
      <div
        className="bag"
      >
        <div
          className="bag-container"
          id="bag-container"
        >
          <span
            className="books-count"
            id="books-count"
          >
            0
          </span>
          <img
            alt="bag"
            id="bag-image"
            onClick={[Function]}
            src="icon.png"
          />
        </div>
      </div>
    </Bag>
  </Connect(Bag)>
</Provider>
`;
